src/poetry/core/masonry/metadata.py

```diff
@@ -107,11 +107,10 @@ def from_package(cls, package: ProjectPackage) -> Metadata:
 
         if package.urls:
             for name, url in package.urls.items():
-                if name.lower() == "homepage" and meta.home_page == url:
+                name_lower = name.lower()
+                if name_lower == "homepage" and meta.home_page == url:
                     continue
-                if name == "repository" and url == package.urls["Repository"]:
-                    continue
-                if name == "documentation" and url == package.urls["Documentation"]:
+                if name_lower == "repository" or name_lower == "documentation":
```

Comment:
I guess the tests have started failing because you have changed the logic

probably want to check "if name is repository and url is repository-url", etc